dss/events/handlers/index.py

```diff
@@ -126,25 +130,110 @@ def add_index_data_to_elasticsearch(bundle_key, index_data, logger) -> None:
 
 
 def create_elasticsearch_index(logger):
+    index_mapping = {
+        "mappings": {
+            "query": {
+                "properties": {
+                    "query": {
+                        "type": "percolator"
+                    }
+                }
+            }
+        }
+    }
     try:
         es_client = ElasticsearchClient.get(logger)
         response = es_client.indices.exists(DSS_ELASTICSEARCH_INDEX_NAME)
-        if response is False:
+        if not response:
             logger.debug(f"Creating new Elasticsearch index: {DSS_ELASTICSEARCH_INDEX_NAME}")
-            response = es_client.indices.create(DSS_ELASTICSEARCH_INDEX_NAME, body=None)
+            response = es_client.indices.create(DSS_ELASTICSEARCH_INDEX_NAME, body=index_mapping)
             logger.debug(f"Index creation response: {json.dumps(response, indent=4)}")
         else:
             logger.debug(f"Using existing Elasticsearch index: {DSS_ELASTICSEARCH_INDEX_NAME}", )
     except Exception as ex:
         logger.critical(f"Unable to create index {DSS_ELASTICSEARCH_INDEX_NAME}  Exception: {ex}")
+        raise
 
 
 def add_data_to_elasticsearch(bundle_id, index_data, logger) -> None:
     try:
         ElasticsearchClient.get(logger).index(index=DSS_ELASTICSEARCH_INDEX_NAME,
                                               doc_type=DSS_ELASTICSEARCH_DOC_TYPE,
                                               id=bundle_id,
-                                              body=json.dumps(index_data, indent=4))
-
+                                              body=json.dumps(index_data),
+                                              refresh=True)
     except Exception as ex:
         logger.error(f"Document not indexed. Exception: {ex}  Index data: {json.dumps(index_data, indent=4)}")
+        raise
+
+
+def find_matching_subscriptions(index_data, logger):
+    percolate_document = {
+        "query": {
+            "percolate": {
+                "field": "query",
+                "document_type": DSS_ELASTICSEARCH_DOC_TYPE,
+                "document": index_data
+            }
+        }
+    }
+    response = ElasticsearchClient.get(logger).search(
+        index=DSS_ELASTICSEARCH_INDEX_NAME,
+        body=percolate_document)
+    logger.debug("Found matching subscription count: %i", len(response['hits']['hits']))
+    subscription_ids = set()
+    for hit in response['hits']['hits']:
```

Comment:
Elasticsearch paginates to 10 results by default. You'll want to make sure you get all results. You can use `for hit in response.scan(): ...` to achieve this.